### PR TITLE
Address some lint and re-render

### PR DESCRIPTION
**Pull request by Cadair**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Replaces #39

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -2,8 +2,10 @@
 {% set version = "67" %}
 {% set root_url = "https://naif.jpl.nasa.gov/pub/naif/misc/toolkit_N%(version)04d/C"|format(version=version | int) %}
 
-{% set platform_segment = "PC_Linux_GCC_64bit" %}                                      # [linux64 or aarch64]
-{% set sha256 = "60a95b51a6472f1afe7e40d77ebdee43c12bb5b8823676ccc74692ddfede06ce" %}  # [linux64 or aarch64]
+{% set file_ext = "tar.Z" %}  # [not win]
+{% set file_ext = "zip" %}  # [win]
+{% set platform_segment = "PC_Linux_GCC_64bit" %}                                      # [linux64 or aarch64 or ppc64le]
+{% set sha256 = "60a95b51a6472f1afe7e40d77ebdee43c12bb5b8823676ccc74692ddfede06ce" %}  # [linux64 or aarch64 or ppc64le]
```
Comment (Cadair) [inline recipe/meta.yaml]: I assume not having a `platform_segment` variable for `ppc64le` was what was at the root of #38 :shrug:

**Comment by Cadair** [comment]
The macos failure looks like it's beyond my pay grade. @AndrewAnnex @astrojuanlu got any thoughts?

File: recipe/meta.yaml (hunk 1 above)
Comment (AndrewAnnex) [reply recipe/meta.yaml]: ppc64le is for powerpc and is not supported, aarch64 is already supported

File: recipe/meta.yaml [hunk 2]
```diff
@@ -16,20 +18,18 @@ package:
...
   patches:
     - 0003_remove_m64_linux.patch  # [linux64 or aarch64]
 
 build:
-  number: 6
+  number: 7
 
 requirements:
   build:
+    - {{ stdlib("c") }}
```
Comment (Cadair) [inline recipe/meta.yaml]: @AndrewAnnex This stdlib change is something the linter picked up which is probably worth migrating to anyway irrespective of the aarch64/ppcle64 stuff.

**Comment by AndrewAnnex** [comment]
okay now that I have more of a grasp of what's happening I am going to work with this PR instead. I decided to remove the power pc stuff as it's not important to support/ spice doesn't support it anyways. I also included what I think can work as a fix for the osx test, although it may take a few more tries to get right

File: recipe/meta.yaml [hunk 3]
```diff
@@ -16,25 +18,24 @@ package:
   version: {{ version }}
 
 source:
-  fn: {{ name }}.tar.Z  # [not win]
-  fn: {{ name }}.zip  # [win]
-  url: {{ root_url }}/{{ platform_segment }}/packages/{{ name }}.tar.Z  # [not win]
-  url: {{ root_url }}/{{ platform_segment }}/packages/{{ name }}.zip  # [win]
+  url: {{ root_url }}/{{ platform_segment }}/packages/{{ name }}.{{ file_ext }}
```
Comment (jakirkham) [inline recipe/meta.yaml:21]:
```suggestion
  {% if platform_segment is defined %}
  url: {{ root_url }}/{{ platform_segment }}/packages/{{ name }}.{{ file_ext }}
  {% endif %}
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -16,25 +18,24 @@ package:
...
   sha256: {{ sha256 }}
```
Comment (jakirkham) [inline recipe/meta.yaml:22]:
```suggestion
  {% if sha256 is defined %}
  sha256: {{ sha256 }}
  {% endif %}
```

**Review by jakirkham — COMMENTED** [review]
Left some comments below to address the bot migration issue: https://github.com/conda-forge/cspice-feedstock/issues/38

**Pull request merged**